x/projects/types/project.go

```diff
@@ -10,15 +14,36 @@ func ProjectIndex(subscriptionAddress string, projectName string) string {
 	return subscriptionAddress + "-" + projectName
 }
 
-func CreateProject(subscriptionAddress string, projectName string) Project {
+func CreateProject(subscriptionAddress string, projectName string) (Project, error) {
+	if !validateProjectName(projectName) {
+		return Project{}, fmt.Errorf("project name must be ASCII and cannot contain \",\". Name: %s", projectName)
+	}
+
 	return Project{
-		Index:        ProjectIndex(subscriptionAddress, projectName),
-		Subscription: subscriptionAddress,
-		Description:  "",
-		ProjectKeys:  []ProjectKey{},
-		Policy:       Policy{},
-		UsedCu:       0,
+		Index:              ProjectIndex(subscriptionAddress, projectName),
+		Subscription:       subscriptionAddress,
+		Description:        "",
+		ProjectKeys:        []ProjectKey{},
+		AdminPolicy:        Policy{},
+		SubscriptionPolicy: Policy{},
+		UsedCu:             0,
+	}, nil
+}
+
+func validateProjectName(projectName string) bool {
+	if strings.Contains(projectName, ",") || !isASCII(projectName) {
+		return false
 	}
+	return true
+}
+
+func isASCII(s string) bool {
```

Comment:
put this function in common, ask oren laadan about this, he might have already made a function like this

Reply:
done (created a file in common types called ascii.go). Also moved ASCII function that Oren wrote to ascii.go